### PR TITLE
[WIP]Add retry and wait time to get pod object in test_drain_mcg_pod_node

### PR DESCRIPTION
**Pull request by tiffanyn108**
Fixed issue [# 6963](https://github.com/red-hat-storage/ocs-ci/issues/6963)
Added retry loop and wait time to get god object.

### CODE REVIEW

**Comment by tiffanyn108** [comment]
> PR validation
> 
> Cluster Name: Cluster Configuration: PR Test Suite: tier3 PR Test Path: tests/manage/mcg/test_mcg_resources_disruptions.py Additional Test Params: OCP VERSION: 4.12 OCS VERSION: 4.12 tested against branch: master
> 
> [Job UNSTABLE](https://url.corp.redhat.com/a26067d) (some or all tests failed).

Failure due to issue https://bugzilla.redhat.com/show_bug.cgi?id=1903573 while running "test_db_scc" test.
"test_drain_mcg_pod_node" is passed as expected.

**Comment by tiffanyn108** [comment]
> tests/manage/mcg/test_mcg_resources_disruptions.py

The failure due to SCC (Security Context Constraint).  This is fixed in[ PR# 6811](https://github.com/red-hat-storage/ocs-ci/pull/6811))

File: tests/manage/mcg/test_mcg_resources_disruptions.py [hunk 1]
```diff
@@ -132,19 +135,44 @@ def test_scale_down_rgw(self, scale_down_to):
...
         """
-        # Retrieve the relevant pod object
-        pod_obj = pod.Pod(
-            **pod.get_pods_having_label(
-                label=self.labels_map[pod_to_drain],
-                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
-            )[0]
-        )
+        loop_cnt = 0
+        while True:
```
Comment (sagihirshfeld) [inline tests/manage/mcg/test_mcg_resources_disruptions.py:149]: We should use utils.py::TimeoutSampler or retry.py::retry instead for brevity, better maintainability and readability

**Review by sagihirshfeld — CHANGES_REQUESTED** [review]
The original issue only happens after the draining of the node at line 179, so we should only be retrying while retrieving the pod_obj at line 185.

**Pull request closed without merging**